core/optaplanner-core-impl/src/test/java/org/optaplanner/core/impl/constructionheuristic/placer/entity/PlacementAssertions.java

```diff
@@ -9,9 +9,9 @@
 import org.optaplanner.core.impl.heuristic.move.Move;
 import org.optaplanner.core.impl.heuristic.selector.move.generic.ChangeMove;
 
-public abstract class AbstractEntityPlacerTest {
```

Comment:
I'm not too keen on deleting these abstract parent classes.

I know that they've been mostly empty, but sharing code is not their only value - how about code structure? Until this PR, we know what the selector factory tests were, what the score definition tests were, etc. And now, they all blend into the crowd.

I am not going to block the PR for this, as I really can't tell if this is important or not, but I do wonder if we're losing something here.

Reply:
You have a valid point and after thinking about it carefully and doing some investigation, I disagree.

> Until this PR, we know what the selector factory tests were, what the score definition tests were, etc.

We still know it even after this PR.
- The score definition tests are all in the `org.optaplanner.core.impl.score.buildin` package.
- The entity/value placer tests are _now_ all in the `org.optaplanner.core.impl.constructionheuristic.placer.entity` package.
- The selector factory tests *not in the same pacakge* but they are all named `*SelectorFactoryTest`.

Good naming practices and following the package structure laid out by the `src/main/java` code are the right tools to organize our tests in my opinion.

The removed abstract classes did not help with the code organization because this practice was not applied consistently:
- The `PooledEntityPlacerFactoryTest` did not extend the `AbstractEntityPlacerTest`. (I found it while deduplicating the `buildHeuristicConfigPolicy()` method.)
- The `HardSoftScoreDefinitionTest` was the only child of the `AbstractScoreDefinitionTest`.
- There are selector factory tests that did not extend the `AbstractSelectorFactoryTest`.

So, in my opinion, using abstract classes to tag similar tests is not useful and we don't lose anything by removing them.

I'll leave some time for @ge0ffrey to share his opinion and I'll merge this after that.

Reply:
No strong opinion from my side.

If it empty now (its builder methods are better at another place?), I guess its natural to delete it?